toolchain/check/testdata/impl/no_prelude/import_builtin_call.carbon

```diff
@@ -0,0 +1,1219 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/check/testdata/impl/no_prelude/import_builtin_call.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/check/testdata/impl/no_prelude/import_builtin_call.carbon
+
+// --- generic_impl.carbon
+
+library "[[@TEST_NAME]]";
+
+interface Add {
+  fn Op[self: Self](other: Self) -> Self;
+}
+
+fn IntLiteral() -> type = "int_literal.make_type";
+fn Int(n: IntLiteral()) -> type = "int.make_type_signed";
+
+class MyInt(N:! IntLiteral()) {
+  adapt Int(N);
+}
+
+impl forall [N:! IntLiteral()] MyInt(N) as Add {
+  fn Op[self: Self](other: Self) -> Self = "int.sadd";
+}
+
+fn Double[N:! IntLiteral()](x: MyInt(N)) -> MyInt(N) {
+  return x.(Add.Op)(x);
+}
+
+// --- use_generic_impl.carbon
+
+library "[[@TEST_NAME]]";
+
+import library "generic_impl";
+
+// TODO: The call below crashes if this one is not present because the generics
+// get imported in a bad order.
+fn LocalDouble(x: MyInt(64)) -> MyInt(64) {
+  return x.(Add.Op)(x);
+}
+
+fn CallImportedDouble(n: MyInt(64)) -> MyInt(64) {
+  return Double(n);
+}
+
+// --- convert_symbolic.carbon
+
+library "[[@TEST_NAME]]";
+
+fn IntLiteral() -> type = "int_literal.make_type";
+fn Int(n: IntLiteral()) -> type = "int.make_type_signed";
+fn ToLiteral(n: Int(32)) -> IntLiteral() = "int.convert_checked";
+fn FromLiteral(n: IntLiteral()) -> Int(32) = "int.convert_checked";
+
+fn Make(N:! Int(32)) -> Int(ToLiteral(N)) { return Make(N); }
+
+class MyInt {
```

Comment:
It would be a bit easier to match SemIR to code if this were renamed to be different from the other `class MyInt(N)`.